compiler/rustc_middle/src/ty/context.rs

```diff
@@ -447,6 +447,9 @@ impl<'tcx> TyCtxt<'tcx> {
     pub fn feed_local_crate(self) -> TyCtxtFeed<'tcx, CrateNum> {
         TyCtxtFeed { tcx: self, key: LOCAL_CRATE }
     }
+    pub fn feed_def_id(self, key: LocalDefId) -> TyCtxtFeed<'tcx, LocalDefId> {
```

Comment:
We could special case this to be a `feed_type_of_anon_const`, ensured by assertions to only be used for the specific cases we want.

I'll write up a hackmd with a plan to use the regular feeding after solving the issues around iterating over body owners if said iteration or later logic may create more body owners